strawberryfields/api/devicespec.py

```diff
@@ -15,62 +15,66 @@
 This module contains a class that represents the specifications of
 a device available via the API.
 """
-from collections.abc import Sequence
-import re
+from typing import Sequence, Mapping, Any
 
 import blackbird
 from blackbird.error import BlackbirdSyntaxError
 
 import strawberryfields as sf
 from strawberryfields.compilers import Ranges
-from strawberryfields.tdm.tdmprogram import TDMProgram
 
 
 class DeviceSpec:
     """The specifications for a specific hardware device.
 
     Args:
-        target (str): name of the target hardware device
         spec (dict): dictionary representing the raw device specification.
-            This dictionary should contain the following key-value pairs:
+            This dictionary must contain the following key-value pairs:
 
-            - layout (str): string containing the Blackbird circuit layout
+            - layout (str): string containing the Blackbird or XIR circuit layout
```

Comment:
I think it would be more accurate to state
```suggestion
            - layout (str): string containing the Blackbird circuit layout or XIR manifest
```